### PR TITLE
Stricten user import schema

### PR DESCRIPTION
**Pull request by pkong-ds**
ref DEV-1482

Some directions I thought of, but only implemented the checked ones
- [x] add `minLength: 1` to required string fields
- [ ] add regex to some expected fields, like email, phone number, birthdate
- [ ] add enum to expected fields, like zoneinfo, locale

ref https://docs.authgear.com/how-to-guide/user-management/import-users-using-user-import-api#input-format

@louischan-oursky see if you expected more strict rules? thanks 🙏

### CODE REVIEW

**Review by louischan-oursky — CHANGES_REQUESTED** [review]
In pkg/lib/authn/stdattrs/validate.go, there are the actual schema that the std attr module will use to validate the standard attributes. We can borrow the schemas there.

We must pay attention to one thing. validation.SchemaBuilder is just a map, so we must not mutate the schema in stdattrs. I am saying this because the schema used in userimport is slightly different from the schemas in stdattrs. In particular, the schema in userimport allows null.

So my suggestion is

1. Make the SchemaBuilder* in stdattrs private. To avoid external modules import them and mutate them accidentally.
2. Expose methods to make new SchemaBuilder in stdattrs. So that the logic still live in stdattrs.
3. Call the newly exposed methods in userimport to validate more strict.

**Comment by pkong-ds** [comment]
@louischan-oursky updated, thanks!

File: pkg/util/validation/schema_builder.go [hunk 1]
```diff
@@ -150,3 +150,19 @@ func (b SchemaBuilder) ToSimpleSchema() *SimpleSchema {
 	}
 	return NewSimpleSchema(string(bytes))
 }
+
+// This function allow copying the schema builder to a reference
+// Expected usage is to avoid mutating original schema builder
+func (b SchemaBuilder) Copy() SchemaBuilder {
```
Comment (louischan-oursky) [inline pkg/util/validation/schema_builder.go:156]: Clone()

File: pkg/util/validation/schema_builder.go [hunk 2]
```diff
@@ -150,3 +150,19 @@ func (b SchemaBuilder) ToSimpleSchema() *SimpleSchema {
...
+	newB := make(SchemaBuilder)
+	for k, v := range b {
+		vb, ok := v.(SchemaBuilder)
+		if ok {
+			newB[k] = vb.Copy()
```
Comment (louischan-oursky) [inline pkg/util/validation/schema_builder.go:161]: Clone()

File: pkg/util/validation/schema_builder.go [hunk 3]
```diff
@@ -166,3 +166,21 @@ func (b SchemaBuilder) Copy() SchemaBuilder {
 
 	return newB
 }
+
+func (b SchemaBuilder) Nullable() SchemaBuilder {
+	existingTypes := b["type"]
```
Comment (louischan-oursky) [inline pkg/util/validation/schema_builder.go:171]: value, ok := b["type"]

File: pkg/util/validation/schema_builder.go [hunk 4]
```diff
@@ -166,3 +166,21 @@ func (b SchemaBuilder) Copy() SchemaBuilder {
...
+
+	if existingTypes == nil {
```
Comment (louischan-oursky) [inline pkg/util/validation/schema_builder.go:173]: if !ok

File: pkg/util/validation/schema_builder.go [hunk 5]
```diff
@@ -166,3 +166,21 @@ func (b SchemaBuilder) Copy() SchemaBuilder {
 
 	return newB
 }
+
+func (b SchemaBuilder) Nullable() SchemaBuilder {
```
Comment (louischan-oursky) [inline pkg/util/validation/schema_builder.go:170]: AddTypeNull()

Comment (louischan-oursky) [inline pkg/util/validation/schema_builder.go:170]: Move near to `Type()` and `Types()`

File: pkg/util/validation/schema_builder.go [hunk 6]
```diff
@@ -166,3 +166,21 @@ func (b SchemaBuilder) Copy() SchemaBuilder {
...
+	existingTypes := b["type"]
+
+	if existingTypes == nil {
+		b.Type(TypeNull)
```
Comment (louischan-oursky) [inline pkg/util/validation/schema_builder.go:174]: This new method should look like `AddRequired()`

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by pkong-ds** [comment]
Updated, thanks!

File: pkg/lib/userimport/model.go [hunk 7]
```diff
@@ -138,7 +138,7 @@ var nonIdentityAwareStandardAttributeKeys []string = []string{
 	"address",
 }
 
-func mapGetNullable[M ~map[string]interface{}, T constraints.Ordered | ~bool](m M, key string) (*T, bool) {
+func mapGetAddTypeNull[M ~map[string]interface{}, T constraints.Ordered | ~bool](m M, key string) (*T, bool) {
```
Comment (louischan-oursky) [inline pkg/lib/userimport/model.go:141]: Revert this change.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by pkong-ds** [comment]
@louischan-oursky sor, fixed up~

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**